### DISCUSSION TITLE
More options for customizing the zoom control buttons in the editor

### DISCUSSION THREAD

**Discussion by DNin01** [ScratchAddons/ScratchAddons#6865] [Ideas]
I would love to see some additional customization options for the `custom-zoom` addon, specifically for the zoom control buttons. Here are some ideas.

#### Hiding the buttons persistently

Currently, you can have the buttons hide and auto-show on hover, but you can't just remove them completely. Some people may want them to be hidden all the time, especially since you can still zoom in/out by holding <kbd>Ctrl</kbd> while scrolling, or with a trackpad.

#### Combining the buttons

Also known as collapsing. When not in use, the zoom control buttons would be replaced with a single zoom-in icon:

![Zoom in button](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/7efdf684-c563-4a11-a625-bbf80d563efa)

Hovering would reveal the buttons to zoom in, zoom out, and reset the zoom level.

#### Using a slider instead

Why mash buttons when you can use a slider? Lots of software have zoom sliders. And oh, this would help so much with zooming into the costume editor.

Any other ideas on how to improve this addon? What should we implement?

**Comment by WorldLanguages** [comment]
> Hiding the buttons persistently

Seems OK, as long as it's a setting. Maybe if we ever add "dangerous" tags to settings we would use it for this.

> Combining the buttons

Doesn't seem too intuitive to me. I've never seen this UI pattern where only one button is shown and hovering it shows the rest. When did hovering over a button ever do anything? It shouldn't look like 2 buttons are gone.

> Using a slider instead

Really interesting! Can we take inspiration from other software? Maybe image editors?

**Reply by mxmou** [reply]
> > Combining the buttons
> 
> Doesn't seem too intuitive to me. I've never seen this UI pattern where only one button is shown and hovering it shows the rest. When did hovering over a button ever do anything? It shouldn't look like 2 buttons are gone.

Scratch already uses this UI pattern:
![image](https://github.com/ScratchAddons/ScratchAddons/assets/51849865/cc83532f-582e-4bb7-b7f9-ed0d90993e6b)

But it only makes sense when there's one primary action and the other buttons are less important, which isn't the case with zoom controls.
